### DISCUSSION TITLE
Favorite Button

### DISCUSSION THREAD

**Discussion by margeh** [microsoft-search/pnp-modern-search#1733] [Q&A]
Hey folks!

My client wants to have a kind of favorite button in the PnP search result webpart.
The button should trigger an Azure function via HTTP call and should hand over a kind of document id and the user name of the user who clicked the button.

I am very new to PnP and spfx development and need kind of a hint, how to start.

Do I need to implement an custom web component which then I can use in a custom template?
Would this be the correct way to develop such a functionality?

Thanks in advance.

**Comment by wobba** [comment]
Correct, you need to look into creating a custom web component and use that in a custom handlebar template.

https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/

**Reply by margeh** [reply]
Web component ist done, now I want to use it in the card layout.
If I insert the custom web component in the cards layout, the component works, but it breaks the whole card design.

How to achive that?

**Reply by wobba** [reply]
I'm no expert on the ui fabric card components unfortunately. Breaks like in CSS, or breaks functionality?

**Reply by margeh** [reply]
Breaks in functionality
Nothing of the item is loading anymore.

![image](https://user-images.githubusercontent.com/52322054/153055297-c2bb089e-9b8c-433a-b4a0-5aa7c7d20f2c.png)

**Reply by wobba** [reply]
What if you add it below the card?

**Reply by calebmil** [reply]
@wobba or @kasperbolarsen, 

Would using a mgt-file object get us the info needed to flag a document as a favorite for a user?  Any ideas on if there is a Graph Toolkit method to flag a document as a favorite?  I have not had any luck finding one ...
